Libraries/VaahStripe.php

```diff
@@ -0,0 +1,252 @@
+
+        $validate = self::validation($inputs);
+
+        if(isset($validate['status']) && $validate['status'] == 'failed')
+        {
+            return $validate;
+        }
+
+        try{
+            $customer_inputs = [
+                'email' => $inputs['customer']['email'],
+                'name' => $inputs['customer']['name'],
+                'address' => [
+                    'line1' => 'New York',
+                    'country' => 'US'
+                ]
+            ];
+
+            if(isset($inputs['address'])){
+                $customer_inputs['address'] = $inputs['address'];
+            }
+
+            // Set Stripe Secret Key
+
+            $customer = Stripe::customers()
+                ->create($customer_inputs);
+
+            $pay_inputs = $inputs['payment'];
+            $pay_inputs['customer'] = $customer['id'];
+
+
+            $pay_intent = Stripe::paymentIntents()
+                ->create($pay_inputs);
+
+
+            $inputs['card']['number'] = str_replace(
+                '-', '', $inputs['card']['number']
+            );
+
+            $data = [
+                "type" => 'card',
+                "card" => $inputs['card']
+            ];
+
+
+            $payment_method = Stripe::paymentMethods()
+                ->create($data);
+
+            Stripe::paymentIntents()->update($pay_intent['id'], [
+                'payment_method' =>  $payment_method['id']
+            ]);
+
+            $pay_confirm = Stripe::paymentIntents()->confirm(
+                $pay_intent['id'],
+                [
+                    "return_url" => $inputs['return_url']
+                ]
+            );
+
+            $response['status'] = 'success';
+            $response['data'] = $pay_confirm;
+
+        }catch(\Exception $e)
+        {
+            $response['status'] = 'failed';
+            $response['errors'][] = $e->getMessage();
+        }
+
+
+        return $response;
+
+    }
+    //----------------------------------------------------------
+    public function subscribe(Request $request)
+    {
+
+        $inputs = $request->all();
+
+        $validate = self::validation($inputs,'subscription');
+
+        if(isset($validate['status']) && $validate['status'] == 'failed')
+        {
+            return $validate;
+        }
+
+        $pay_inputs = $inputs['payment'];
+
+        $package_price_id = $this->getPriceId($pay_inputs);
+
+        if(is_null($package_price_id) || empty($package_price_id))
+        {
+            $response = [
+                "status" => 'failed',
+                "errors" => ['Plan or plan price for '.$pay_inputs['package']." does not exist"],
+            ];
+            return $response;
+        }
+
+        try{
+            $customer_inputs = [
+                'email' => $inputs['customer']['email'],
+                'name' => $inputs['customer']['name'],
+                'address' => [
+                    'line1' => 'New York',
+                    'country' => 'US'
```

Comment:
this should be a dynamic value which needs to be passed

Reply:
@themodernpk Sir, I have already put condition below this array to add dynamic address

```
$customer_inputs = [
                'email' => $inputs['customer']['email'],
                'name' => $inputs['customer']['name'],
                'address' => [
                    'line1' => 'New York',
                    'country' => 'US'
                ]
            ];

            if($inputs['address']){
                $customer_inputs['address'] = $inputs['address'];
            }
```

I forgot to update in Readme